java-checks/src/main/java/org/sonar/java/checks/security/PasswordEncoderCheck.java

```diff
@@ -0,0 +1,102 @@
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks.security;
+
+import java.util.Arrays;
+import java.util.List;
+import org.sonar.check.Rule;
+import org.sonar.java.matcher.MethodMatcher;
+import org.sonar.java.matcher.MethodMatcherCollection;
+import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
+import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
+import org.sonar.plugins.java.api.tree.MethodInvocationTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
+import org.sonar.plugins.java.api.tree.Tree;
+
+@Rule(key = "S5344")
+public class PasswordEncoderCheck extends IssuableSubscriptionVisitor {
+
+  private static final MethodMatcher JDBC_AUTHENTICATION = MethodMatcher.create()
+    .typeDefinition("org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder")
+    .name("jdbcAuthentication")
+    .withoutParameter();
+
+  private static final MethodMatcher USER_DETAIL_SERVICE = MethodMatcher.create()
+    .typeDefinition("org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder")
+    .name("userDetailsService")
+    .withAnyParameters();
+
+  private static final MethodMatcher PASSWORD_ENCODER_SETTER = MethodMatcher.create()
+    .typeDefinition("org.springframework.security.config.annotation.authentication.configurers.userdetails.AbstractDaoAuthenticationConfigurer")
+    .name("passwordEncoder")
+    .withAnyParameters();
+
+  private static final MethodMatcherCollection UNSAFE_PASSWORD_ENCODERS = MethodMatcherCollection.create(
+    constructorMatcher("org.springframework.security.authentication.encoding.ShaPasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.authentication.encoding.Md5PasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.crypto.password.LdapShaPasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.crypto.password.Md4PasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.crypto.password.MessageDigestPasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.crypto.password.NoOpPasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.crypto.password.StandardPasswordEncoder").withAnyParameters(),
+    constructorMatcher("org.springframework.security.crypto.password.SCryptPasswordEncoder").withAnyParameters()
+  );
+
+  @Override
+  public List<Tree.Kind> nodesToVisit() {
+    return Arrays.asList(Tree.Kind.METHOD, Tree.Kind.NEW_CLASS);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (tree.is(Tree.Kind.NEW_CLASS) && UNSAFE_PASSWORD_ENCODERS.anyMatch(((NewClassTree) tree))) {
```

Comment:
I think we should check if we have semantic information
```
    if (!hasSemantic()) {
      return;
    }
```

Reply:
fixed